### DISCUSSION TITLE
How to revert Solo/Pro-HD to last official firmware?

### DISCUSSION THREAD

**Discussion by robosmith** [GerryDazoo/Slinger#275] [Q&A]
It seems Dish has done a number on my Pro-HD, because trying to log in with my admin password (Slingplayer 1.5.1) no longer works.
I can tell the admin pw I'm using is the last one that worked because if I try anything else, it immediately displays an error.

So I'm assuming I need to fix the trashed firmware with the last official one which I found at: http://autoupdate.sling.com/wbsp/100.07.NA.99/

There are 2 possible avenues with which I am familiar. 
1. Easiest (if it works) is to connect the (apparent) UART serial interface on the board to USB with a cheap adapter available online for <$10
I have found the 4 pin pattern on the board which appears to be the UART interface, but my first effort to read it while booting shows nothing but garbage. It may need pulldown resistors to work.
I was able to upload firmware to a TP Link router using that kind of interface (one pullup resistor), but I had help telling me where to load it in memory.

2. There is a JTAG interface on the board, but all the pin holes are filled in with solder. Not hard to suck them out; I already did one.
From my brief research, it should be possible to upload firmware using JTAG, if the interface is still functional.
I was able to d/l 2 different IDE development kits for the Davinci TMS240 DSP chip from: https://www.ti.com/tool/download/CCSTUDIO/4.1.0.02006
but I have not yet explored the contents. They are large.
Any help from experienced JTAGers would be appreciated.
Thanks.

**Comment by Mikoshi-nyudo** [comment]
Am not an experienced JTAGer or electronics engineer, but from my preliminary investigation I might suggest that one should try to flash first the SbPbProHdFwV2_01_280_HwV1_0_0Master.bin as it contains the entire DSP boot loader with UBL Headers and CPU OS firmware, so it's like Factory shipped. Then it should be possible to use the setup assistant to manually flash an update to the last SbPbProHdFwV2_01_420_HwV1_0_0Current.bin and setup the box again through that assistant.

Hopefully, some more experienced 'experts' will chime in, and figure out the best and proper path in due course.

**Reply by robosmith** [reply]
Great idea. I was unaware of the additional code available in that earlier version. Thanks.
In any case, the prior version should make the box useable again. Esp if the admin password is not updated by the fw flash.

**Comment by GerryDazoo** [comment]
On your UART you may have the wrong baud rate. Or you may be missing the needed level shifters 5V> 
https://www.robotshop.com/ca/en/seeedstudio-rs-232-to-ttl-converter-max3232idr.html?gclid=Cj0KCQiAyMKbBhD1ARIsANs7rEF8DJF3yuyTJunKRH3RP4YB5EvfpsjtkLhQokEfYchIdg4avyyltEgaAkRDEALw_wcB

**Reply by robosmith** [reply]
Thanks for your suggestions.
This is the adapter I am using: https://www.amazon.com/HiLetgo-CP2102-Converter-Adapter-Downloader/dp/B00LODGRV8/ref=sr_1_7_sspa?keywords=uart+usb+adapter&qid=1668378717&sprefix=uart+usb+%2Caps%2C148&sr=8-7-spons&sp_csd=d2lkZ2V0TmFtZT1zcF9tdGY&psc=1

The Pro-HD uses 3.3 v power, and I believe that is what the TP-Link router that I had success with, uses, too.
The adapter is not connected to the 3.3 v on the SB.

I am not sure how to adjust the baud rate, as with the router I just connected it and launched Putty set to COM5 and it immediately showed a text stream. One of the 2 (probably tx) pins needed a pullup resistor to actually read commands sent.

**Reply by GerryDazoo** [reply]
![image](https://user-images.githubusercontent.com/102599361/204100071-d2b0e763-96dd-404c-83c8-106982f1d8f1.png)

Setting baud rate. 
PS usually the 4 pin header is something like V+ tx rx gnd usually you connect all 4 pins to the adapter. to give power to the level shifters.

**Comment by RandomMan719** [comment]
There maybe a possible third option, by doing a factory reset and using the software, to update firmware.  But I don't know how to do it.

**Comment by RocketAlf** [comment]
In the meantime, is there any result for accessing the UART serial interface?

**Comment by RocketAlf** [comment]
I soldered a 14-pin connector for the JTAG interface to my Slingbox Solo. I tried to connect using the Texas Instruments XDS100V2 adapter and Code Composer Studio (tried with v7.3 and v12.1). Unfortunately didn't work. During the connection test I got the error message: " integrity scan-test has failed".
Does anyone here know if and how the JTAG interface can be activated?
